### PR TITLE
fix combining --search and --try-*

### PR DESCRIPTION
**Pull request by boegel**
fix for bug reported in #1371 and #1935

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86 please review

File: easybuild/framework/easyconfig/tweak.py [hunk 1]
```diff
@@ -94,18 +94,15 @@ def tweak(easyconfigs, build_specs, modtool, targetdir=None):
     # keep track of originally listed easyconfigs (via their path)
     listed_ec_paths = [ec['spec'] for ec in easyconfigs]
 
-    # obtain full dependency graph for specified easyconfigs
-    # easyconfigs will be ordered 'top-to-bottom': toolchain dependencies and toolchain first
-    orig_ecs = resolve_dependencies(easyconfigs, modtool, retain_all_deps=True)
```
Comment (wpoely86) [inline easybuild/framework/easyconfig/tweak.py]: why is this gone?

Comment (boegel) [inline easybuild/framework/easyconfig/tweak.py]: this is actually a bug, this line is overwriting the `orig_ecs` defined above in the `if-else`...

**Comment by boegel** [comment]
Thanks for the review @wpoely86!

**Pull request merged**